core/src/main/scala/org/ensime/core/debug/VMEventManager.scala

```diff
@@ -7,37 +7,43 @@ import com.sun.jdi.VMDisconnectedException
 import com.sun.jdi.event.{ ClassPrepareEvent, VMDisconnectEvent, EventQueue }
 import org.slf4j.LoggerFactory
 
-class VMEventManager(val eventQueue: EventQueue, debugManager: ActorRef) extends Thread {
+class VMEventManager(val eventQueue: EventQueue, debugManager: ActorRef) {
 
   val log = LoggerFactory.getLogger("VMEventManager")
 
-  // TODO needs proper stop method with interrupt
-
-  @volatile var finished = false
-  override def run(): Unit = {
-    while (!finished) {
-      try {
-        val eventSet = eventQueue.remove()
-        val it = eventSet.eventIterator()
-        while (it.hasNext) {
-          val evt = it.nextEvent()
-          evt match {
-            case e: VMDisconnectEvent =>
-              finished = true
-            case e: ClassPrepareEvent =>
-              debugManager ! DMClassPrepareEvent(e, eventSet)
-            case _ =>
+  val thread = new Thread() {
+    override def run(): Unit = {
+      while (!this.isInterrupted) {
```

Comment:
I think I would actually keep the finished flag - and then the stop method sets the flag and interrupts.  - that means the other catches just set log something and set the flag.  Feels cleaner to me - but it is stylistic.